### PR TITLE
Fix broken links in 00-introduction.ipynb with nbsphinx 0.9.2

### PR DESCRIPTION
**Pull request by cpoerschke**
**Checklist**

- [ ] closes #xxxx
- [ ] pytest passes
- [ ] tests are included
- [ ] code is well formatted
- [ ] documentation renders correctly

**What does this implement/fix? Explain your changes**

Recent CI runs fail with `Warning, treated as error: /home/runner/work/scikit-survival/scikit-survival/doc/user_guide/00-introduction.ipynb:35:undefined label: /user_guide/00-introduction.ipynb#the-veterans%27-administration-lung-cancer-trial` and experimentally the link on https://scikit-survival.readthedocs.io/en/stable/user_guide/00-introduction.html is broken. The https://scikit-survival.readthedocs.io/en/latest/user_guide/00-introduction.html links work though.

### CODE REVIEW

File: doc/user_guide/00-introduction.ipynb [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "## Table of Contents\n",
     "\n",
     "1. [What is Survival Analysis?](#What-is-Survival-Analysis?)\n",
-    "2. [The Veterans' Administration Lung Cancer Trial](#The-Veterans%27-Administration-Lung-Cancer-Trial)\n",
+    "2. [The Veterans' Administration Lung Cancer Trial](#The-Veterans%E2%80%99-Administration-Lung-Cancer-Trial)\n",
```
Comment (cpoerschke) [inline doc/user_guide/00-introduction.ipynb:24]: CI now says `/home/runner/work/scikit-survival/scikit-survival/doc/user_guide/00-introduction.ipynb:35:undefined label: /user_guide/00-introduction.ipynb#the-veterans%e2%80%99-administration-lung-cancer-trial` but here's another(?) alternative based on https://scikit-survival.readthedocs.io/en/latest/user_guide/00-introduction.html

```suggestion
    "2. [The Veterans' Administration Lung Cancer Trial](#The-Veterans'-Administration-Lung-Cancer-Trial)\n",
```

**Comment by sebp** [comment]
It's strange that CI fails now. For me, the links to on the _stable_ and _latest_ docs work both fine. It might be that a newer version of sphinx (or its dependencies) is used on CI, compared to readthedocs.

**Comment by sebp** [comment]
I can confirm that this is caused by nbsphinx 0.9.2.

**Comment by sebp** [comment]
@cpoerschke Could you please change https://github.com/sebp/scikit-survival/blob/d5fcbc9e06f4824f987b5cf550ce0bf82665f28e/pyproject.toml#L95
to
```
"nbsphinx>=0.9.2",
```

**Pull request merged**